File: LAB 3/SC_CONV_ELLIPSE_POLY.m
clc; clear; close all;

RX = 15;
RY = 8;
CX = 25;
CY = 15;

figure;
hold on;
grid on;
axis equal;
title('Ellipse - Polynomial Method');
xlabel('X-axis');
ylabel('Y-axis');

X_VALS = [];
Y_VALS = [];

for X = CX - RX:CX + RX
    Y_OFFSET = RY * sqrt(1 - ((X - CX) ^ 2) / (RX ^ 2));
    Y1 = round(CY + Y_OFFSET);
    Y2 = round(CY - Y_OFFSET);
    X_VALS = [X_VALS X X];
    Y_VALS = [Y_VALS Y1 Y2];
end

plot(X_VALS, Y_VALS, 'ks', 'MarkerSize', 4, 'MarkerFaceColor', 'k');
plot(CX, CY, 'ro', 'MarkerSize', 8, 'MarkerFaceColor', 'r');

File: LAB 3/Python/SC_CONV_ELLIPSE_POLY.py
import matplotlib.pyplot as plt
import numpy as np

RX = 15
RY = 8
CX, CY = 25, 15

plt.figure()
plt.grid(True)
plt.axis('equal')
plt.title('Ellipse - Polynomial Method')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')

X_VALS = []
Y_VALS = []

for X in range(CX - RX, CX + RX + 1):
    Y_OFFSET = RY * np.sqrt(1 - ((X - CX) ** 2) / (RX ** 2))
    Y1 = round(CY + Y_OFFSET)
    Y2 = round(CY - Y_OFFSET)
    X_VALS.append(X)
    X_VALS.append(X)
    Y_VALS.append(Y1)
    Y_VALS.append(Y2)

plt.plot(X_VALS, Y_VALS, 'ks', markersize=4, markerfacecolor='k')
plt.plot(CX, CY, 'ro', markersize=8, markerfacecolor='r')
plt.show()
